### PR TITLE
Added branch component

### PR DESCRIPTION
**Pull request by CarloPalinckx**
### This PR:

proposed release: `minor`

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable)

**Adds** ✨
- A Branch component

### CODE REVIEW

File: src/components/Branch/story.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import React from 'react';
+import Text from '../Text';
+
+storiesOf('Branch', module).add('Default', () => {
+    return (
+        <Branch
+            condition={boolean('condition', true)}
+            ifTrue={(children): JSX.Element => <Text>This is shown when condition is true. {children}</Text>}
+            ifFalse={(children): JSX.Element => (
+                <Text>And this is will be shown when condition is false. {children}</Text>
```
Comment (hborghols) [inline src/components/Branch/story.tsx]: typo (this is will)

File: src/components/Branch/story.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            )}
+        >
+            This line is shared wether the condition is true or not.
```
Comment (hborghols) [inline src/components/Branch/story.tsx]: wether -> whether

File: src/components/Branch/story.tsx (hunk 1 above)
Comment (woutsluiter) [reply src/components/Branch/story.tsx]: This is Will
image

Comment (CarloPalinckx) [reply src/components/Branch/story.tsx]: 😂

Comment (hborghols) [reply src/components/Branch/story.tsx]: 🤣

**Pull request merged**